contracts/evm/contracts/xcall/CallService.sol

```diff
@@ -116,122 +111,221 @@ contract CallService is IBSH, ICallService, IFeeManage, Initializable {
         bytes memory _rollback,
         string[] memory sources,
         string[] memory destinations
-    ) external payable override returns (
-        uint256
-    ) {
+    ) external payable override returns (uint256) {
         return _sendCallMessage(_to, _data, _rollback, sources, destinations);
     }
 
-
     function sendCallMessage(
         string memory _to,
         bytes memory _data,
         bytes memory _rollback
-    ) external payable override returns (
-        uint256
-    ) {
+    ) external payable override returns (uint256) {
         string[] memory src;
         string[] memory dst;
         return _sendCallMessage(_to, _data, _rollback, src, dst);
     }
 
-     function _sendCallMessage(
+    event CallMessageReq(string from,
+        string to,
+        uint256 sn,
+        int messageType,
+        bytes data);
+
+    function sendCall(
         string memory _to,
-        bytes memory _data,
-        bytes memory _rollback,
-        string[] memory sources,
-        string[] memory destinations
-    ) internal returns (
-        uint256
-    ) {
-        // check if caller is a contract or rollback data is null in case of EOA
-        require(msg.sender.code.length > 0 || _rollback.length == 0, "RollbackNotPossible");
-
-        // check size of payloads to avoid abusing
-        require(_rollback.length <= MAX_ROLLBACK_SIZE, "MaxRollbackSizeExceeded");
-
-        bool needResponse = _rollback.length > 0;
-        (string memory netTo, string memory dstAccount) = _to.parseNetworkAddress();
-        string memory from = nid.networkAddress(msg.sender.toString());
+        bytes memory _data
+    ) public payable returns (uint256) {
+        address caller = msg.sender;
+        Types.XCallEnvelope memory envelope = _data.decodeXCallEnvelope();
         uint256 sn = getNextSn();
-        int256 msgSn = 0;
-        if (needResponse) {
-            requests[sn] = Types.CallRequest(msg.sender, netTo, sources, _rollback, false);
-            msgSn = int256(sn);
-        }
-        Types.CSMessageRequest memory reqMsg = Types.CSMessageRequest(
-            from, dstAccount, sn, needResponse, _data, destinations);
-        bytes memory _msg = reqMsg.encodeCSMessageRequest();
+        Types.ProcessResult memory result = preProcessMessage(
+            sn,
+            _to,
+            envelope
+        );
+        string memory from = nid.networkAddress(caller.toString());
+
+        (string memory netTo, string memory dstAccount) = _to
+            .parseNetworkAddress();
+
+        emit CallMessageReq(from, _to, sn, envelope.messageType, result.data);
```

Comment:
why was this event added?

Reply:
Oh, it was used while testing, will removed that